extensions-core/kafka-extraction-namespace/src/main/java/io/druid/query/lookup/KafkaLookupExtractorFactory.java

```diff
@@ -98,220 +97,207 @@ public String fromBytes(byte[] bytes)
 
   @JsonCreator
   public KafkaLookupExtractorFactory(
-      @JacksonInject NamespaceExtractionCacheManager cacheManager,
-      @JsonProperty("kafkaTopic") final String kafkaTopic,
-      @JsonProperty("kafkaProperties") final Map<String, String> kafkaProperties,
-      @JsonProperty("connectTimeout") @Min(0) long connectTimeout,
-      @JsonProperty("injective") boolean injective
+          @JacksonInject NamespaceExtractionCacheManager cacheManager,
+          @JsonProperty("kafkaTopic") final String kafkaTopic,
+          @JsonProperty("kafkaProperties") final Map<String, String> kafkaProperties,
+          @JsonProperty("connectTimeout") @Min(0) long connectTimeout,
+          @JsonProperty("injective") boolean injective
+  )
+  {
+    this(cacheManager, kafkaTopic, kafkaProperties, connectTimeout, injective, new ConsumerConnectorFactory.Zookeeper());
+  }
+
+  KafkaLookupExtractorFactory(
+          NamespaceExtractionCacheManager cacheManager,
+          final String kafkaTopic,
+          final Map<String, String> kafkaProperties,
+          long connectTimeout,
+          boolean injective,
+          ConsumerConnectorFactory consumerConnectorFactory
   )
   {
     this.kafkaTopic = Preconditions.checkNotNull(kafkaTopic, "kafkaTopic required");
     this.kafkaProperties = Preconditions.checkNotNull(kafkaProperties, "kafkaProperties required");
-    executorService = MoreExecutors.listeningDecorator(Execs.singleThreaded(
-        "kafka-factory-" + kafkaTopic + "-%s",
-        Thread.MIN_PRIORITY
-    ));
     this.cacheManager = cacheManager;
     this.connectTimeout = connectTimeout;
     this.injective = injective;
+    this.consumerConnectorFactory = consumerConnectorFactory;
+    this.executorService = MoreExecutors.listeningDecorator(Execs.singleThreaded(
+            "kafka-factory-" + kafkaTopic + "-%s",
+            Thread.MIN_PRIORITY
+    ));
   }
 
-  public KafkaLookupExtractorFactory(
-      NamespaceExtractionCacheManager cacheManager,
-      String kafkaTopic,
-      Map<String, String> kafkaProperties
-  )
-  {
-    this(cacheManager, kafkaTopic, kafkaProperties, 0, false);
+  void init() {
+    map = cacheManager.getCacheMap(factoryId);
   }
 
-  public String getKafkaTopic()
+  @Override
+  public boolean start()
   {
-    return kafkaTopic;
-  }
 
-  public Map<String, String> getKafkaProperties()
-  {
-    return kafkaProperties;
-  }
+    if(!advanceState(State.NEW, State.STARTING)) {
+      LOG.warn("Already started, not starting again. Current state: [%s]", currentState.get());
+      return currentState.get().equals(State.RUNNING);
+    }
 
-  public long getConnectTimeout()
-  {
-    return connectTimeout;
-  }
+    // TODO: if current state is STARTING, block until a terminal state is reached?
```

Comment:
Can there ever be two concurrent invocations of `start()`? What is the expected behaviour in that case?